### ISSUE TITLE
remove event-entity from loop-entity

### ISSUE THREAD

**Issue by BitRoit** [SkriptLang/Skript#5396]
### Skript/Server Version

```markdown
[Skript] Server Version: git-Purpur-1824 (MC: 1.19.2)
[Skript] Skript Version: 2.6.4
[Skript] Installed Skript Addons: None
[Skript] Installed dependencies: 
[Skript]  - Vault v1.7.3-b131
[Skript]  - WorldGuard v7.0.8-beta-01+cbb2ba7
```

### Bug Description

"remove event-entity from loop-entity" removes player from loop-entity but it also clears event-entity items and creates an error message in console.

### Expected Behavior

It should just remove player from loop-entity

### Steps to Reproduce

I added remove player from loop-entity and if player is sprinting:
```v
options:
    Cooldown: 3 seconds
    Speed: 2

function Dash(p: player):
    set {_u} to {_p}'s uuid
    if {dash.last::%{_u}%} is set:
        set {wait} to difference between {dash.last::%{_u}%} and now
    else:
        set {wait} to {@Cooldown}
    if {wait} >= {@Cooldown}:
        set {dash.last::%{_u}%} to now
        if {_p} is on ground:
            push {_p} forwards at speed {@Speed}
        else:
            push {_p} forwards at speed {@Speed} / 2.5
#	give 1 of shield named "§cT§6a§er§ac§9z§ba §cZ§6r§ey§aw§9u" with lore "&7Kliknij prawym przyciskiem myszy podczas biegu aby wykonać zryw!" to player
on right click:
	if player is sprinting:
		give 1 of shield named "§cT§6a§er§ac§9z§ba §cZ§6r§ey§aw§9u" with lore "&7Kliknij prawym przyciskiem myszy podczas biegu aby wykonać zryw!" to player
		if player's tool is shield named "§cT§6a§er§ac§9z§ba §cZ§6r§ey§aw§9u" with lore "&7Kliknij prawym przyciskiem myszy podczas biegu aby wykonać zryw!":
			Dash(player)
			if {wait} >= {@Cooldown}:
				damage player's tool by 1
				loop all entities in radius 2 of player:
					remove event-entity from loop-entity
#					remove player from loop-entity
					damage loop-entity by 3 hearts
```

### Agreement

- [x] I have read the guidelines above and affirm I am following them with this report.

**Comment by TheLimeGlass** [comment]
What's the error message it provided?

**Comment by BitRoit** [comment]
[15:39:47 ERROR]: #!#!
[15:39:47 ERROR]: #!#! [Skript] Severe Error:
[15:39:47 ERROR]: #!#!
[15:39:47 ERROR]: #!#! Something went horribly wrong with Skript.
[15:39:47 ERROR]: #!#! This issue is NOT your fault! You probably can't fix it yourself, either.
[15:39:47 ERROR]: #!#! It looks like you are using some plugin(s) that alter how Skript works (addons).
[15:39:47 ERROR]: #!#! Here is full list of them:
[15:39:47 ERROR]: #!#! skript-reflect v2.3 (https://github.com/TPGamesNL/skript-reflect) MorkazSk v1.3 skript-gui v1.3 (https://github.com/APickledWalrus/skript-gui) SkBee v2.7.2 (https://github.com/ShaneBeee/SkBee) SkQuery v4.1.7 MythicSkriptAddon v0.99e-SNAPSHOT
[15:39:47 ERROR]: #!#! We could not identify which of those are specially related, so this might also be Skript issue.
[15:39:47 ERROR]: #!#! You should try disabling those plugins one by one, trying to find which one causes it.
[15:39:47 ERROR]: #!#! If the error doesn't disappear even after disabling all listed plugins, it is probably Skript issue.
[15:39:47 ERROR]: #!#! In that case, you will be given instruction on how should you report it.
[15:39:47 ERROR]: #!#! On the other hand, if the error disappears when disabling some plugin, report it to author of that plugin.
[15:39:47 ERROR]: #!#! Only if the author tells you to do so, report it to Skript's issue tracker.
[15:39:47 ERROR]: #!#!
[15:39:47 ERROR]: #!#! Stack trace:
[15:39:47 ERROR]: #!#! java.lang.NullPointerException: Cannot invoke "org.bukkit.inventory.ItemStack.getAmount()" because "item" is null
[15:39:47 ERROR]: #!#!     at org.bukkit.craftbukkit.v1_19_R1.inventory.CraftInventory.removeItem(CraftInventory.java:376)
[15:39:47 ERROR]: #!#!     at org.bukkit.craftbukkit.v1_19_R1.inventory.CraftInventory.removeItem(CraftInventory.java:359)
[15:39:47 ERROR]: #!#!     at Skript-2.6.4.jar//ch.njol.skript.classes.data.DefaultChangers$1.change(DefaultChangers.java:100)
[15:39:47 ERROR]: #!#!     at Skript-2.6.4.jar//ch.njol.skript.classes.data.DefaultChangers$1.change(DefaultChangers.java:50)
[15:39:47 ERROR]: #!#!     at Skript-2.6.4.jar//ch.njol.skript.lang.util.SimpleExpression.change(SimpleExpression.java:240)
[15:39:47 ERROR]: #!#!     at Skript-2.6.4.jar//ch.njol.skript.effects.EffChange.execute(EffChange.java:282)
[15:39:47 ERROR]: #!#!     at Skript-2.6.4.jar//ch.njol.skript.lang.Effect.run(Effect.java:50)
[15:39:47 ERROR]: #!#!     at Skript-2.6.4.jar//ch.njol.skript.lang.TriggerItem.walk(TriggerItem.java:61)
[15:39:47 ERROR]: #!#!     at Skript-2.6.4.jar//ch.njol.skript.lang.TriggerItem.walk(TriggerItem.java:89)
[15:39:47 ERROR]: #!#!     at Skript-2.6.4.jar//ch.njol.skript.lang.Trigger.execute(Trigger.java:56)
[15:39:47 ERROR]: #!#!     at Skript-2.6.4.jar//ch.njol.skript.SkriptEventHandler.check(SkriptEventHandler.java:142)
[15:39:47 ERROR]: #!#!     at Skript-2.6.4.jar//ch.njol.skript.SkriptEventHandler.access$000(SkriptEventHandler.java:53)
[15:39:47 ERROR]: #!#!     at Skript-2.6.4.jar//ch.njol.skript.SkriptEventHandler$PriorityListener.lambda$new$0(SkriptEventHandler.java:71)
[15:39:47 ERROR]: #!#!     at co.aikar.timings.TimedEventExecutor.execute(TimedEventExecutor.java:76)
[15:39:47 ERROR]: #!#!     at org.bukkit.plugin.RegisteredListener.callEvent(RegisteredListener.java:70)
[15:39:47 ERROR]: #!#!     at org.bukkit.plugin.SimplePluginManager.callEvent(SimplePluginManager.java:678)
[15:39:47 ERROR]: #!#!     at org.bukkit.craftbukkit.v1_19_R1.event.CraftEventFactory.callPlayerInteractEvent(CraftEventFactory.java:545)
[15:39:47 ERROR]: #!#!     at org.bukkit.craftbukkit.v1_19_R1.event.CraftEventFactory.callPlayerInteractEvent(CraftEventFactory.java:502)
[15:39:47 ERROR]: #!#!     at org.bukkit.craftbukkit.v1_19_R1.event.CraftEventFactory.callPlayerInteractEvent(CraftEventFactory.java:497)
[15:39:47 ERROR]: #!#!     at org.bukkit.craftbukkit.v1_19_R1.event.CraftEventFactory.callPlayerInteractEvent(CraftEventFactory.java:492)
[15:39:47 ERROR]: #!#!     at org.bukkit.craftbukkit.v1_19_R1.event.CraftEventFactory.callPlayerInteractEvent(CraftEventFactory.java:488)
[15:39:47 ERROR]: #!#!     at net.minecraft.server.network.PlayerConnection.a(PlayerConnection.java:2109)
[15:39:47 ERROR]: #!#!     at net.minecraft.network.protocol.game.PacketPlayInBlockPlace.a(PacketPlayInBlockPlace.java:32)
[15:39:47 ERROR]: #!#!     at net.minecraft.network.protocol.game.PacketPlayInBlockPlace.a(PacketPlayInBlockPlace.java:8)
[15:39:47 ERROR]: #!#!     at net.minecraft.network.protocol.PlayerConnectionUtils.lambda$ensureRunningOnSameThread$0(PlayerConnectionUtils.java:51)
[15:39:47 ERROR]: #!#!     at net.minecraft.server.TickTask.run(TickTask.java:18)
[15:39:47 ERROR]: #!#!     at net.minecraft.util.thread.IAsyncTaskHandler.d(IAsyncTaskHandler.java:153)
[15:39:47 ERROR]: #!#!     at net.minecraft.util.thread.IAsyncTaskHandlerReentrant.d(IAsyncTaskHandlerReentrant.java:24)
[15:39:47 ERROR]: #!#!     at net.minecraft.server.MinecraftServer.b(MinecraftServer.java:1368)
[15:39:47 ERROR]: #!#!     at net.minecraft.server.MinecraftServer.d(MinecraftServer.java:185)
[15:39:47 ERROR]: #!#!     at net.minecraft.util.thread.IAsyncTaskHandler.x(IAsyncTaskHandler.java:126)
[15:39:47 ERROR]: #!#!     at net.minecraft.server.MinecraftServer.bh(MinecraftServer.java:1345)
[15:39:47 ERROR]: #!#!     at net.minecraft.server.MinecraftServer.x(MinecraftServer.java:1338)
[15:39:47 ERROR]: #!#!     at net.minecraft.util.thread.IAsyncTaskHandler.c(IAsyncTaskHandler.java:136)
[15:39:47 ERROR]: #!#!     at net.minecraft.server.MinecraftServer.w(MinecraftServer.java:1316)
[15:39:47 ERROR]: #!#!     at net.minecraft.server.MinecraftServer.v(MinecraftServer.java:1204)
[15:39:47 ERROR]: #!#!     at net.minecraft.server.MinecraftServer.lambda$spin$1(MinecraftServer.java:310)
[15:39:47 ERROR]: #!#!     at java.base/java.lang.Thread.run(Thread.java:1589)
[15:39:47 ERROR]: #!#!
[15:39:47 ERROR]: #!#! Version Information:
[15:39:47 ERROR]: #!#!   Skript: 2.6.4 (latest)
[15:39:47 ERROR]: #!#!     Flavor: skriptlang-github
[15:39:47 ERROR]: #!#!     Date: 17:14:25.041218900
[15:39:47 ERROR]: #!#!   Bukkit: 1.19.2-R0.1-SNAPSHOT
[15:39:47 ERROR]: #!#!   Minecraft: 1.19.2
[15:39:47 ERROR]: #!#!   Java: 19.0.1 (Java HotSpot(TM) 64-Bit Server VM 19.0.1+10-21)
[15:39:47 ERROR]: #!#!   OS: Windows 10 amd64 10.0
[15:39:47 ERROR]: #!#!
[15:39:47 ERROR]: #!#! Server platform: Paper
[15:39:47 ERROR]: #!#!
[15:39:47 ERROR]: #!#! Current node: null
[15:39:47 ERROR]: #!#! Current item: remove (event-entity)->org.bukkit.inventory.Inventory from loop-entity
[15:39:47 ERROR]: #!#! Current trigger: right click (rightclick) (Sus.sk, line 278)
[15:39:47 ERROR]: #!#!
[15:39:47 ERROR]: #!#! Thread: Server thread
[15:39:47 ERROR]: #!#!
[15:39:47 ERROR]: #!#! Language: english
[15:39:47 ERROR]: #!#! Link parse mode: DISABLED
[15:39:47 ERROR]: #!#!
[15:39:47 ERROR]: #!#! End of Error.
[15:39:47 ERROR]: #!#!

I'm trying to remove the entity that uses the shield from loop-entity.

**Comment by BitRoit** [comment]
remove {_p} from loop-entity doesn't work aswell

on right click:
	if player is sprinting:
		give 1 of shield named "§cT§6a§er§ac§9z§ba §cZ§6r§ey§aw§9u" with lore "&7Kliknij prawym przyciskiem myszy podczas biegu aby wykonać zryw!" to player
		if player's tool is shield named "§cT§6a§er§ac§9z§ba §cZ§6r§ey§aw§9u" with lore "&7Kliknij prawym przyciskiem myszy podczas biegu aby wykonać zryw!":
			Dash(player)
			if {wait} >= {@Cooldown}:
				damage player's tool by 1
				loop all entities in radius 5 of player:
					remove {_p} from loop-entity
					remove loop-entity from {_p}
					damage loop-entity by 3 hearts
				push player backwards at speed 1.5


options:
    Cooldown: 3 seconds
    Speed: 2

function Dash(p: player):
    set {_u} to {_p}'s uuid
    if {dash.last::%{_u}%} is set:
        set {wait} to difference between {dash.last::%{_u}%} and now
    else:
        set {wait} to {@Cooldown}
    if {wait} >= {@Cooldown}:
        set {dash.last::%{_u}%} to now
        if {_p} is on ground:
            push {_p} forwards at speed {@Speed}
        else:
            push {_p} forwards at speed {@Speed} / 2.5

**Comment by BitRoit** [comment]
"remove event-entity from loop-entity" removes player from loop-entity but it also clears event-entity items and creates an error message in console.

**Comment by AyhamAl-Ali** [comment]
Just realised _(Thanks to APickledWalrus)_ you're trying to remove entities that are using shield from the looped entities, the way you're trying to do that is wrong, you should be doing it like this
```v
loop all entities in radius 2 of player: # should actually be 'all players'
	delete loop-entity # will delete the entity (you can use kill loop-entity as well)
```

**Comment by BitRoit** [comment]
This is not what I want to do, I tried to make it so that for a while I would get a list of entities around the player, but not the player himself who clicked the right mouse button.

**Comment by AyhamAl-Ali** [comment]
Can you be more specific about what you're trying to do?

Quoted
> I'm trying to remove the entity that uses the shield from loop-entity.

If this is what you're trying to do then the code I sent above is correct (see the edited version now - added a filter for the player himself)

if you're trying to do something else then explain it more please.

**Comment by BitRoit** [comment]
I tried to make a shield that would allow the player, when right-clicking, to rush at enemies (pushing the player at speed 2) and if it detected any in its path, it would hurt them and push the player back at speed 2.5

**Comment by BitRoit** [comment]
Do I need any addons for that?

**Comment by BitRoit** [comment]
And it would also consume some durability from the shield

**Comment by BitRoit** [comment]
?

**Comment by BitRoit** [comment]
Hmm

**Comment by TheLimeGlass** [comment]
Fixed in 2.7.0-beta2
